### PR TITLE
added webvtt exporter (i.e. compiler)

### PR DESCRIPTION
**Pull request by goatandsheep**
exporter is a more descriptive word than compiler, but could be mess up es6 imports #11

### CODE REVIEW

**Comment by goatandsheep** [comment]
What can I do to speed up time before this can be merged?

File: lib/compiler.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  this.message = message;
+  this.error = error;
+}
+CompilerError.prototype = Object.create(Error.prototype);
+
+function compile (input) {
+
+  if (!input) {
+    throw new CompilerError('Input must be non-null');
+  }
```
Comment (osk) [inline lib/compiler.js:18]: I want spaces between blocks. eslint rules weren't catching but I just updated `master`, adding the `padding-line-between-statements`

File: lib/compiler.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+function compileCue (cue) {
+  // TODO: check for malformed JSON
+  if (typeof cue !== 'object') {
+    throw new CompilerError('Cue malformed: not of type object');
+  }
+  if (typeof cue.identifier !== 'string' &&
+    typeof cue.identifier !== 'number' &&
+    cue.identifier !== null) {
+    throw new CompilerError(`Cue malformed: identifier value is not a string.
+    ${JSON.stringify(cue)}`);
```
Comment (osk) [inline lib/compiler.js:62]: This looks weird aligned like this, rather keep it all in one line, adding a `/* eslint-disable-line */` to the end

File: lib/compiler.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+ * @returns {object} cue Cue object with start, end, text and styles.
+ *                       Null if it's a note
+ */
+function compileCue (cue) {
+  // TODO: check for malformed JSON
+  if (typeof cue !== 'object') {
+    throw new CompilerError('Cue malformed: not of type object');
+  }
+  if (typeof cue.identifier !== 'string' &&
+    typeof cue.identifier !== 'number' &&
```
Comment (osk) [inline lib/compiler.js]: I want the expressions aligned, skimming it looks like the expression is not part of the `if`.
Did a quick check of eslint rules but didn't find anything to enforce this

File: lib/compiler.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+  let output = '';
+  if (cue.identifier.length > 0) {
+    output += `${cue.identifier}\n`;
+  }
+
+  const startTimestamp = convertTimestamp(cue.start);
+  const endTimestamp = convertTimestamp(cue.end);
+
+  output += `${startTimestamp} --> ${endTimestamp} ${cue.styles}\n${cue.text}`;
```
Comment (osk) [inline lib/compiler.js]: The space between `${endTimestamp} ${cue.styles}` causes a redundant trailing space if there are no `styles`, it should check if `cue.styles` is set and then adding it + a space

**Review by osk — CHANGES_REQUESTED** [review]
Looks good! Some comment style changes that need to be changed, I've updated my `master` with a new rule so you can rebase and fix.

I wonder if there should be a new integration test suite, i.e. that makes sure something that's parsed and compiled looks like the original again, e.g.:

```javascript
compile(parse(input)).should.equal(input);
```

Also, could you add code examples to `README.md` about how to use this in praxis?

File: lib/compiler.js (hunk 1 above)
Comment (goatandsheep) [reply lib/compiler.js:18]: I'm so happy you updated eslint. Made the updates really easy

**Comment by goatandsheep** [comment]
I made the changes

**Review by osk — APPROVED** [review]

**Comment by osk** [comment]
Nice!

**Comment by osk** [comment]
v1.2 published with this

**Comment by goatandsheep** [comment]
I forgot to add the update to the README about usage. I'll add testing config and update README in next PR.

**Pull request merged**